### PR TITLE
x11-wm/stumpwm: Add version 22.05

### PR DESCRIPTION
**Pull request by Thuna-Cing**
I have been using version 22.05 for a while now (2-3 weeks), and I have not seen any issues. I imagine the older versions should also work but I do not intend to test them.

### CODE REVIEW

**Comment by Thuna-Cing** [comment]
Alright, I amended the commit and resolved some of the suggestions.

File: x11-wm/stumpwm/stumpwm-22.05.ebuild [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+KEYWORDS="~amd64 ~x86"
+IUSE="contrib doc emacs"
+
+RESTRICT="strip mirror"
+
+RDEPEND="dev-lisp/alexandria
+		dev-lisp/cl-ppcre
+		dev-lisp/clx
+		>=dev-lisp/sbcl-1.3.0
+		emacs? ( >=app-editors/emacs-23.1:* app-emacs/slime )"
```
[the remark this answers is not in the document]
Comment (Thuna-Cing) [reply x11-wm/stumpwm/stumpwm-22.05.ebuild]: I am actually in favor of removing the `emacs` flag all together. It doesn't seem all that necessary to me, and forcing people to use `app-emacs/slime` when there are alternatives is also not great.

[the remark this answers is not in the document]
Comment (Thuna-Cing) [reply x11-wm/stumpwm/stumpwm-22.05.ebuild]: Nevermind actually. The site regen for emacs needs to happen whether `app-emacs/slime` is there or not, so the flag `emacs` needs to be a thing.  I'll just remove the `app-emacs/slime` dependency instead.

**Review by juippis — COMMENTED** [review]
I can not compile this. Looks like I'm hitting https://github.com/stumpwm/stumpwm/issues/994 but this patch doesn't apply cleanly anymore. Seems like it's merged after 22.05, so could you do a snapshot-based ebuild, picking some latest good commit to base the ebuild release on? Here's an example of a simple snapshot release in Gentoo:
https://gitweb.gentoo.org/repo/gentoo.git/tree/media-libs/rlottie/rlottie-0.2_p20210405.ebuild
Base the version as release_p<date-of-commit>

Add a `MY_COMMIT` variable, modify `SRC_URI` and `S` as needed.

File: x11-wm/stumpwm/stumpwm-22.05.ebuild [hunk 2]
```diff
@@ -0,0 +1,76 @@
+# Copyright 1999-2022 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+inherit common-lisp-3 autotools desktop elisp-common xdg-utils
+
+DESCRIPTION="Stumpwm is a Window Manager written entirely in Common Lisp"
```
Comment (juippis) [inline x11-wm/stumpwm/stumpwm-22.05.ebuild:8]:
```suggestion
DESCRIPTION="A Window Manager written entirely in Common Lisp"
```
or
```suggestion
DESCRIPTION="Window Manager written entirely in Common Lisp"
```
We don't repeat `PN` here.

**Comment by Thuna-Cing** [comment]
Alright, the commit I picked to base the ebuild on was the last related bug fix I could find.  The ebuild emerged fine on my end.
If the two previous commits are fine, they should be rebased onto the first commit.

**Review by juippis — APPROVED** [review]
I can't build docs, that seems totally broken. I'll make a bug report about it. Thanks!

I'll do some minor fixes in a follow-up commit.

**Pull request closed without merging**